lib/std/http/status.zig

```diff
@@ -0,0 +1,203 @@
+        non_authoritative_info = 203, // RFC7231, Section 6.3.4
+        no_content = 204, // RFC7231, Section 6.3.5
+        reset_content = 205, // RFC7231, Section 6.3.6
+        partial_content = 206, // RFC7233, Section 4.1
+        multi_status = 207, // RFC4918
+        already_reported = 208, // RFC5842
+        im_used = 226, // RFC3229
+
+        multiple_choice = 300, // RFC7231, Section 6.4.1
+        moved_permanently = 301, // RFC7231, Section 6.4.2
+        found = 302, // RFC7231, Section 6.4.3
+        see_other = 303, // RFC7231, Section 6.4.4
+        not_modified = 304, // RFC7232, Section 4.1
+        use_proxy = 305, // RFC7231, Section 6.4.5
+        temporary_redirect = 307, // RFC7231, Section 6.4.7
+        permanent_redirect = 308, // RFC7538
+
+        bad_request = 400, // RFC7231, Section 6.5.1
+        unauthorized = 401, // RFC7235, Section 3.1
+        payment_required = 402, // RFC7231, Section 6.5.2
+        forbidden = 403, // RFC7231, Section 6.5.3
+        not_found = 404, // RFC7231, Section 6.5.4
+        method_not_allowed = 405, // RFC7231, Section 6.5.5
+        not_acceptable = 406, // RFC7231, Section 6.5.6
+        proxy_auth_required = 407, // RFC7235, Section 3.2
+        request_timeout = 408, // RFC7231, Section 6.5.7
+        conflict = 409, // RFC7231, Section 6.5.8
+        gone = 410, // RFC7231, Section 6.5.9
+        length_required = 411, // RFC7231, Section 6.5.10
+        precondition_failed = 412, // RFC7232, Section 4.2][RFC8144, Section 3.2
+        payload_too_large = 413, // RFC7231, Section 6.5.11
+        uri_too_long = 414, // RFC7231, Section 6.5.12
+        unsupported_media_type = 415, // RFC7231, Section 6.5.13][RFC7694, Section 3
+        range_not_satisfiable = 416, // RFC7233, Section 4.4
+        expectation_failed = 417, // RFC7231, Section 6.5.14
+        teapot = 418, // RFC 7168, 2.3.3
+        misdirected_request = 421, // RFC7540, Section 9.1.2
+        unprocessable_entity = 422, // RFC4918
+        locked = 423, // RFC4918
+        failed_dependency = 424, // RFC4918
+        too_early = 425, // RFC8470
+        upgrade_required = 426, // RFC7231, Section 6.5.15
+        precondition_required = 428, // RFC6585
+        too_many_requests = 429, // RFC6585
+        header_fields_too_large = 431, // RFC6585
+        unavailable_for_legal_reasons = 451, // RFC7725
+
+        internal_server_error = 500, // RFC7231, Section 6.6.1
+        not_implemented = 501, // RFC7231, Section 6.6.2
+        bad_gateway = 502, // RFC7231, Section 6.6.3
+        service_unavailable = 503, // RFC7231, Section 6.6.4
+        gateway_timeout = 504, // RFC7231, Section 6.6.5
+        http_version_not_supported = 505, // RFC7231, Section 6.6.6
+        variant_also_negotiates = 506, // RFC2295
+        insufficient_storage = 507, // RFC4918
+        loop_detected = 508, // RFC5842
+        not_extended = 510, // RFC2774
+        network_authentication_required = 511, // RFC6585
+
```

Comment:
This should be an extensible enum
```suggestion
+        _,

```

Reply:
no, that's what `Status.unknown` is for

Reply:
That would mean you get an ABI break every time a new status code is added.
Get rid of the `unknown` enum and use an extensible enum

Reply:
why would it cause a break? there's no reason why someone would switch on it.

`Status.unknown` is a union option that is a struct so that if someone tries to parse a non standard code, the phrase text is not lost. this prevents needing multiple status types